### ISSUE TITLE
The "Fully-qualify things in all namespaces" commit killed examples

### ISSUE THREAD

**Issue by marcinkaszynski** [dropbox/djinni#78]
That commit misdirects the compiler to look for names like `sort_order`, `ItemList` etc in global namespace instead of `textsort`.

**Comment by marcinkaszynski** [comment]
Here's the output from `make example_android`.  I had to make two small changes to the build file to make it work in my environment, but that looks irrelevant to the results.

```shell
[@trurl djinniMU master] git diff
diff --git a/example/android/app/build.gradle b/example/android/app/build.gradle
index ec0b0d1..d005c22 100644
--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -14,7 +14,7 @@ android {
     buildTypes {
         release {
             runProguard false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+//            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
     sourceSets {
@@ -32,7 +32,7 @@ dependencies {

 task ndkBuild(type: Exec) {
     String MainDirectory = System.getProperty("user.dir") + '/app/'
-    File ndkDir = project.getPlugins().getPlugin('android').ndkFolder
+    File ndkDir = project.getPlugins().getPlugin('android').sdkHandler.getNdkFolder()
     if (ndkDir == null) {
         def gradle_project_root = project.rootProject.rootDir
         throw new GradleException("NDK is not configured. Make sure there is a local.properties " +

[@trurl djinniMU master] make example_android
cd example/android/ && ./gradlew app:assembleDebug
:app:preBuild
:app:compileDebugNdk UP-TO-DATE
:app:preDebugBuild
:app:checkDebugManifest
:app:prepareDebugDependencies
:app:compileDebugAidl UP-TO-DATE
:app:compileDebugRenderscript UP-TO-DATE
:app:generateDebugBuildConfig UP-TO-DATE
:app:generateDebugAssets UP-TO-DATE
:app:mergeDebugAssets UP-TO-DATE
:app:generateDebugResValues UP-TO-DATE
:app:generateDebugResources UP-TO-DATE
:app:mergeDebugResources UP-TO-DATE
:app:processDebugManifest UP-TO-DATE
:app:processDebugResources UP-TO-DATE
:app:generateDebugSources UP-TO-DATE
:app:ndkBuild
make[1]: Entering directory `/Users/marcin/checkout/collectively/djinni/example/android/app'
[arm64-v8a] Compile++      : textsort_jni <= NativeSortItems.cpp
In file included from ../../../example/generated-src/jni/NativeSortItems.hpp:7:0,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:4:
../../../example/generated-src/cpp/sort_items.hpp:18:25: error: '::sort_order' has not been declared
     virtual void sort(::sort_order order, const ::ItemList & items) = 0;
                         ^
../../../example/generated-src/cpp/sort_items.hpp:18:51: error: 'ItemList' in namespace '::' does not name a type
     virtual void sort(::sort_order order, const ::ItemList & items) = 0;
                                                   ^
../../../example/generated-src/cpp/sort_items.hpp:20:28: error: '::SortItems' has not been declared
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                            ^
../../../example/generated-src/cpp/sort_items.hpp:20:28: note: suggested alternative:
../../../example/generated-src/cpp/sort_items.hpp:14:7: note:   'textsort::SortItems'
 class SortItems {
       ^
../../../example/generated-src/cpp/sort_items.hpp:20:28: error: '::SortItems' has not been declared
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                            ^
../../../example/generated-src/cpp/sort_items.hpp:20:28: note: suggested alternative:
../../../example/generated-src/cpp/sort_items.hpp:14:7: note:   'textsort::SortItems'
 class SortItems {
       ^
../../../example/generated-src/cpp/sort_items.hpp:20:39: error: template argument 1 is invalid
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                                       ^
../../../example/generated-src/cpp/sort_items.hpp:20:84: error: '::TextboxListener' has not been declared
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                                                                                    ^
../../../example/generated-src/cpp/sort_items.hpp:20:84: note: suggested alternative:
../../../example/generated-src/cpp/sort_items.hpp:12:7: note:   'textsort::TextboxListener'
 class TextboxListener;
       ^
../../../example/generated-src/cpp/sort_items.hpp:20:84: error: '::TextboxListener' has not been declared
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                                                                                    ^
../../../example/generated-src/cpp/sort_items.hpp:20:84: note: suggested alternative:
../../../example/generated-src/cpp/sort_items.hpp:12:7: note:   'textsort::TextboxListener'
 class TextboxListener;
       ^
../../../example/generated-src/cpp/sort_items.hpp:20:101: error: template argument 1 is invalid
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                                                                                                     ^
In file included from ../../../example/generated-src/jni/NativeTextboxListener.hpp:7:0,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:8:
../../../example/generated-src/cpp/textbox_listener.hpp:14:33: error: 'ItemList' in namespace '::' does not name a type
     virtual void update(const ::ItemList & items) = 0;
                                 ^
In file included from ../../../example/generated-src/jni/NativeSortItems.cpp:8:0:
../../../example/generated-src/jni/NativeTextboxListener.hpp:25:22: error: 'virtual void djinni_generated::NativeTextboxListener::JavaProxy::update(const textsort::ItemList&)' marked override, but does not override
         virtual void update(const ::textsort::ItemList & items) override;
                      ^
../../../example/generated-src/jni/NativeSortItems.cpp: In function 'void djinni_generated::Java_com_dropbox_textsort_SortItems_00024CppProxy_native_1sort(JNIEnv*, jobject, jlong, jobject, jobject)':
../../../example/generated-src/jni/NativeSortItems.cpp:32:35: error: no matching function for call to 'textsort::SortItems::sort(textsort::sort_order&, textsort::ItemList&)'
         ref->sort(c_order, c_items);
                                   ^
../../../example/generated-src/jni/NativeSortItems.cpp:32:35: note: candidate is:
In file included from ../../../example/generated-src/jni/NativeSortItems.hpp:7:0,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:4:
../../../example/generated-src/cpp/sort_items.hpp:18:18: note: virtual void textsort::SortItems::sort(int, const int&)
     virtual void sort(::sort_order order, const ::ItemList & items) = 0;
                  ^
../../../example/generated-src/cpp/sort_items.hpp:18:18: note:   no known conversion for argument 1 from 'textsort::sort_order' to 'int'
../../../example/generated-src/jni/NativeSortItems.cpp: In function '_jobject* djinni_generated::Java_com_dropbox_textsort_SortItems_createWithListener(JNIEnv*, jobject, jobject)':
../../../example/generated-src/jni/NativeSortItems.cpp:42:107: error: no matching function for call to 'textsort::SortItems::create_with_listener(std::shared_ptr<textsort::TextboxListener>&)'
         std::shared_ptr<::textsort::SortItems> cr = ::textsort::SortItems::create_with_listener(c_listener);
                                                                                                           ^
../../../example/generated-src/jni/NativeSortItems.cpp:42:107: note: candidate is:
In file included from ../../../example/generated-src/jni/NativeSortItems.hpp:7:0,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:4:
../../../example/generated-src/cpp/sort_items.hpp:20:41: note: static int textsort::SortItems::create_with_listener(const int&)
     static std::shared_ptr<::SortItems> create_with_listener(const std::shared_ptr<::TextboxListener> & listener);
                                         ^
../../../example/generated-src/cpp/sort_items.hpp:20:41: note:   no known conversion for argument 1 from 'std::shared_ptr<textsort::TextboxListener>' to 'const int&'
In file included from /Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/libs/arm64-v8a/include/bits/c++allocator.h:33:0,
                 from /Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/allocator.h:46,
                 from /Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/memory:63,
                 from ../../../support-lib/jni/djinni_support.hpp:21,
                 from ../../../example/generated-src/jni/NativeSortItems.hpp:6,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:4:
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = djinni_generated::NativeTextboxListener::JavaProxy; _Args = {_jobject*&}; _Tp = djinni_generated::NativeTextboxListener::JavaProxy]':
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/alloc_traits.h:253:4:   required from 'static std::_Require<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type> std::allocator_traits<_Alloc>::_S_construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = djinni_generated::NativeTextboxListener::JavaProxy; _Args = {_jobject*&}; _Alloc = std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>; std::_Require<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type> = void]'
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/alloc_traits.h:399:57:   required from 'static decltype (_S_construct(__a, __p, (forward<_Args>)(std::allocator_traits::construct::__args)...)) std::allocator_traits<_Alloc>::construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = djinni_generated::NativeTextboxListener::JavaProxy; _Args = {_jobject*&}; _Alloc = std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>; decltype (_S_construct(__a, __p, (forward<_Args>)(std::allocator_traits::construct::__args)...)) = <type error>]'
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/shared_ptr_base.h:516:38:   required from 'std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {_jobject*&}; _Tp = djinni_generated::NativeTextboxListener::JavaProxy; _Alloc = std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]'
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/ext/new_allocator.h:127:4:   required from 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = std::_Sp_counted_ptr_inplace<djinni_generated::NativeTextboxListener::JavaProxy, std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, (__gnu_cxx::_Lock_policy)2u>; _Args = {const std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, _jobject*&}; _Tp = std::_Sp_counted_ptr_inplace<djinni_generated::NativeTextboxListener::JavaProxy, std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, (__gnu_cxx::_Lock_policy)2u>]'
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/alloc_traits.h:253:4:   required from 'static std::_Require<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type> std::allocator_traits<_Alloc>::_S_construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = std::_Sp_counted_ptr_inplace<djinni_generated::NativeTextboxListener::JavaProxy, std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, (__gnu_cxx::_Lock_policy)2u>; _Args = {const std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, _jobject*&}; _Alloc = std::allocator<std::_Sp_counted_ptr_inplace<djinni_generated::NativeTextboxListener::JavaProxy, std::allocator<djinni_generated::NativeTextboxListener::JavaProxy>, (__gnu_cxx::_Lock_policy)2u> >; std::_Require<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type> = void]'
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/alloc_traits.h:399:57:   [ skipping 5 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/shared_ptr.h:604:42:   required from 'std::shared_ptr<_Tp1> std::make_shared(_Args&& ...) [with _Tp = djinni_generated::NativeTextboxListener::JavaProxy; _Args = {_jobject*&}]'
../../../support-lib/jni/djinni_support.hpp:317:57:   required from 'static std::pair<std::shared_ptr<void>, _jobject*> djinni::JavaProxyCache<T>::factory(jobject) [with T = djinni_generated::NativeTextboxListener::JavaProxy; jobject = _jobject*]'
../../../support-lib/jni/djinni_support.hpp:329:70:   required from 'static std::shared_ptr<_Tp1> djinni::JavaProxyCache<T>::get(jobject) [with T = djinni_generated::NativeTextboxListener::JavaProxy; jobject = _jobject*]'
../../../support-lib/jni/djinni_support.hpp:536:60:   required from 'std::shared_ptr<_Tp1> djinni::JniInterface<I, Self>::_getJavaProxy(jobject) const [with S = djinni_generated::NativeTextboxListener; <template-parameter-2-2> = djinni_generated::NativeTextboxListener::JavaProxy; I = textsort::TextboxListener; Self = djinni_generated::NativeTextboxListener; jobject = _jobject*]'
../../../support-lib/jni/djinni_support.hpp:483:29:   required from 'std::shared_ptr<_Tp1> djinni::JniInterface<I, Self>::_fromJava(JNIEnv*, jobject) const [with I = textsort::TextboxListener; Self = djinni_generated::NativeTextboxListener; JNIEnv = _JNIEnv; jobject = _jobject*]'
../../../example/generated-src/jni/NativeTextboxListener.hpp:17:188:   required from here
/Users/marcin/Library/Android/ndk-r10d/sources/cxx-stl/gnu-libstdc++/4.9/include/ext/new_allocator.h:127:4: error: invalid new-expression of abstract class type 'djinni_generated::NativeTextboxListener::JavaProxy'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from ../../../example/generated-src/jni/NativeSortItems.cpp:8:0:
../../../example/generated-src/jni/NativeTextboxListener.hpp:22:11: note:   because the following virtual functions are pure within 'djinni_generated::NativeTextboxListener::JavaProxy':
     class JavaProxy final : djinni::JavaProxyCacheEntry, public ::textsort::TextboxListener {
           ^
In file included from ../../../example/generated-src/jni/NativeTextboxListener.hpp:7:0,
                 from ../../../example/generated-src/jni/NativeSortItems.cpp:8:
../../../example/generated-src/cpp/textbox_listener.hpp:14:18: note: 	virtual void textsort::TextboxListener::update(const int&)
     virtual void update(const ::ItemList & items) = 0;
                  ^
make[1]: *** [/Users/marcin/checkout/collectively/djinni/example/android/app//obj/local/arm64-v8a/objs/textsort_jni/example/generated-src/jni/NativeSortItems.o] Error 1
make[1]: Leaving directory `/Users/marcin/checkout/collectively/djinni/example/android/app'
:app:ndkBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:ndkBuild'.
> Process 'command '/Users/marcin/Library/Android/ndk-r10d/ndk-build'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 6.89 secs
make: *** [example_android] Error 1
[@trurl djinniMU master]```

**Comment by j4cbo** [comment]
Sorry about that. Should be fixed by 30e73aebcd446502d1a952326471a8890bdec97b.

**Comment by marcinkaszynski** [comment]
Yup, that worked. Thanks!
